### PR TITLE
Allow users to upload role versions

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #3529
https://pulp.plan.io/issues/3529

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -94,6 +94,63 @@ Look at the new Repository Version created
...
+        "namespace": "pulp"
+    }
+
+
+Create a ``role version`` from the Role and Artifact
+-----------------------------------------------------
+
+Create a content unit and point it to your Artifact and Role
+
+``$ http POST http://localhost:8000/api/v3/content/ansible/roles/3965b515-53a0-4667-a540-a76714a903d4/versions/ version=0.0.1 artifact="http://localhost:8000/api/v3/artifacts/7d39e3f6-535a-4b6e-81e9-c83aa56aa19e/"``
```
Comment (bmbouter) [inline README.rst]: Can these use exported variables so it can be copy/paste? That is such a nice format.

File: README.rst [hunk 2]
```diff
@@ -94,6 +94,40 @@ Look at the new Repository Version created
...
+
+Create a content unit and point it to your Artifact and Role
+
+``$ export CONTENT_HREF=$(http POST ${ROLE_HREF}versions/ version=0.0.1 artifact=$ARTIFACT_HREF | jq -r '._href')``
+
+
+Add content to repository ``foo``
+---------------------------------
+
+``$ http POST $REPO_HREF'versions/' add_content_units:="[\"$CONTENT_HREF\"]"``
```
Comment (bmbouter) [inline README.rst]: I had trouble with this command. It says:

```
(pulp) [vagrant@pulp3 ~]$ http POST $REPO_HREF'versions/' add_content_units:="[\"$CONTENT_HREF\"]"
HTTP/1.1 400 Bad Request
Allow: GET, POST, HEAD, OPTIONS
Content-Type: application/json
Date: Mon, 23 Apr 2018 20:00:09 GMT
Server: WSGIServer/0.2 CPython/3.6.2
Vary: Accept

[
    "URI http://localhost:8000/api/v3/content/ansible/roles/069d5cf7-bc77-46b7-a2cf-95b390f03548/versions/3f7bacb8-4b62-4208-bbeb-e74226bf3984/ is not a valid content."
]
```

Comment (daviddavis) [reply README.rst]: Here's the fix:

https://github.com/pulp/pulp/pull/3468

Comment (bmbouter) [reply README.rst]: That other PR got merged so I resumed my testing.

**Pull request merged**